lib/streamlit/DeltaGenerator.py

```diff
@@ -1841,6 +1843,75 @@ def slider(
             current_value = current_value[0] if single_value else current_value
         return current_value if single_value else tuple(current_value)
 
+    @_with_element
+    def file_uploader(self, element, label, type=['*']):
+
+        """Display a file uploader widget.
+
+        Parameters
+        ----------
+        label : str or None
+            A short label explaining to the user what this file uploader is for.
+        type : string[] or None
+            Array of allowed extensions. (['.png', '.jpg'])
+            Default value: ['*']
+
+        Returns
+        -------
+        str 
+            The id of stored file or empty if no one file is loaded.
+
+        Examples
+        --------
+        >>> file = st.file_uploader("Upload a image", type=([".png"]))
+        >>> if file:
+        >>>     file_bytes = st.file_reader(file)
+        >>>     st.image(file_bytes)
+
+        """
+
+        element.file_uploader.label = label
+        element.file_uploader.type[:] = type
+        element.file_uploader.max_upload_size = config.get_option("server.maxUploadSize")
+        ui_value = _get_widget_ui_value("file_uploader", element)
+        
+        return ui_value if ui_value is not None else ""
+
+
+    def file_reader(self, file_id):
```

Comment:
I don't understand: what is this? Is this another widget type? Or just some leftover code?